vector/src/main/java/im/vector/app/features/settings/notifications/VectorSettingsNotificationPreferenceFragment.kt

```diff
@@ -116,11 +120,48 @@ class VectorSettingsNotificationPreferenceFragment @Inject constructor(
             }
         }
 
+        bindEmailNotifications()
         refreshBackgroundSyncPrefs()
 
         handleSystemPreference()
     }
 
+    private fun bindEmailNotifications() {
+        val emails = session.getEmailsWithPushInformation()
+        findPreference<VectorPreferenceCategory>(VectorPreferences.SETTINGS_EMAIL_NOTIFICATION_CATEGORY_PREFERENCE_KEY)?.let { category ->
+            if (emails.isEmpty()) {
+                val vectorPreference = VectorPreference(requireContext())
+                vectorPreference.title = resources.getString(R.string.settings_notification_emails_no_emails)
+                category.addPreference(vectorPreference)
+                vectorPreference.setOnPreferenceClickListener {
+                    interactionListener?.navigateToEmailAndPhoneNumbers()
```

Comment:
If I go this way and add an email to the account, when coming back here, the screen is not refreshed. Maybe force a refresh in onResume()

Reply:
another option is to make use of `getThreePidsLive` which also has the benefit of updating the emails whilst in the screen, so far it's working well :crossed_fingers:

Reply:
far better, thanks!